common/src/main/java/com/revenuecat/purchases/common/verification/SigningManager.kt

```diff
@@ -2,18 +2,24 @@ package com.revenuecat.purchases.common.verification
 
 import android.util.Base64
 import com.revenuecat.purchases.common.errorLog
+import com.revenuecat.purchases.common.networking.Endpoint
 import com.revenuecat.purchases.common.networking.HTTPResult
+import com.revenuecat.purchases.common.networking.RCHTTPStatusCodes
 import com.revenuecat.purchases.strings.NetworkStrings
 import java.security.SecureRandom
 
 class SigningManager(
-    private val signatureVerifier: SignatureVerifier
+    val signatureVerificationMode: SignatureVerificationMode
 ) {
     private companion object {
         const val NONCE_BYTES_SIZE = 12
         const val SALT_BYTES_SIZE = 16
     }
 
+    fun shouldVerifyEndpoint(endpoint: Endpoint): Boolean {
+        return endpoint.supportsSignatureValidation && signatureVerificationMode.shouldVerify
```

Comment:
Moved this logic here, to move logic away from HTTPClient.

Reply:
👍🏻